### PR TITLE
Set wishlist form key

### PR DESCRIPTION
**Pull request by deepjoshi94**
### Description
Adding product to wishlist when user is not logged in.
If user clicks on add to wishlist button magento will redirect to login page if user is not logged in, once user is logged in system will redirect to wishlist page for the user but that particular product will not be added to the wishlist.

Fixed the above mentioned issue.

### Manual testing scenarios
1. Add product to wishlist when user is not logged in.
2. System will redirect user to login page.
3. Once user login, user will be redirected to wishlist page.
4. That particular product will be added to wishlist page for the user.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=17468) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=17468) before we can accept your contribution.<br/><hr/>**Deep Joshi** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

**Comment by magento-engcom-team** [comment]
Hi @deepjoshi94. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

File: app/code/Magento/Wishlist/Helper/Data.php
Comment (orlangur) [inline app/code/Magento/Wishlist/Helper/Data.php]: Looks like this logic eliminates the whole point in form key, isn't it? As after such change CSRF is possible with automatic generation of form key.

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Wishlist/Helper/Data.php
Comment (deepjoshi94) [reply app/code/Magento/Wishlist/Helper/Data.php]: @orlangur Reason of setting form key is to take selected product from guest session to customer session, when user logs in.

**Review by orlangur — CHANGES_REQUESTED** [review]
Code alignment looks awkward, please fix it amending initial commit.

Nowadays all changes must be applied to `2.3-develop` first (see https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules). Please prepare a new pull request, this one should be `on hold` until the latter is merged.

**Comment by deepjoshi94** [comment]
@orlangur I have aligned code as requested.
My commit id for reference: **4918fbe**

**Comment by orlangur** [comment]
@deepjoshi94 I don't see a new PR targeting `2.3-develop`.

**Comment by sivaschenko** [comment]
Hi @deepjoshi94 I am closing this pull request due to inactivity, feel free to reopen if you'd like to continue the progress on this contribution.

**Pull request closed without merging**
